sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConstants.cs

```diff
@@ -23,5 +23,14 @@ internal class AmqpMessageConstants
         internal const string TimeSpanName = AmqpConstants.Vendor + ":timespan";
         internal const string UriName = AmqpConstants.Vendor + ":uri";
         internal const string DateTimeOffsetName = AmqpConstants.Vendor + ":datetime-offset";
+        /// <summary>
+        /// User property key representing deadletter reason, when a message is received from a deadletter subqueue of an entity.
```

Comment:
Is it a user property? This is a system property key, set by the broker.

Reply:
It is in the user properties bag since the user can actually set the reason/error when deadlettering.

Reply:
It's not set directly, unlike user set properties/headers. The value is passed into dead-letter method and set by the broker.

Also, it's a single usecase. Messages dead letter for other reasons and messages are neither controlled nor set by users but the broker.

Reply:
Correct, but the doc comments were intended to indicate where the property key lives. It wasn't meant to imply that a user can just change it whenever. I can clarify.